cli/src/commands/extensions/mod.rs

```diff
@@ -0,0 +1,113 @@
+pub mod extension;
+
+use std::{collections::HashSet, convert::TryFrom, path::PathBuf};
+
+use crate::commands::{CommandResult, CommandValue, ExitCode};
+pub use extension::*;
+use log::{error, info};
+
+use anyhow::Result;
+use clap::{arg, ArgMatches, Command, ValueHint};
+
+pub fn command<'a>() -> Command<'a> {
+    Command::new("extension")
+        .about("Run extensions")
+        .subcommand(
+            Command::new("add")
+                .about("Install extension")
+                .arg(arg!([PATH]).required(true).value_hint(ValueHint::FilePath)),
+        )
+        .subcommand(
+            Command::new("remove")
+                .about("Uninstall extension")
+                .arg(arg!([NAME]).required(true)),
+        )
+        .subcommand(Command::new("list").about("List installed extensions"))
+}
```

Comment:
I wanted to suggest the opposite -- I feel like separating the command definitions from their modules makes it harder to maintain (and test -- we may want  to add smaller unit tests that only go through `clap` at some point and I feel like their own modules would be a better place for that) and think it would be better to have a leaner `app.rs`.